### ISSUE TITLE
Remove image in FileManager in use by RealBlog

### ISSUE THREAD

**Issue by BillyV2018** [cmsimple-xh/cmsimple-xh#635]
The FileManager will correctly detect when an image is in use on one of the pages. 
But it will remove an image used in one of the blog Articles, and that is a pity.

**Comment by cmb69** [comment]
> An API would have to be created in the file browser for this, and the plugins would also all have to be revised.
> I see little chance of this happening.

Right. And there is the additional hurdle that plugins like Realblog allow to insert plugin calls. This is not even catered to be the search of Realblog, since that could slow down the search considerably (nobody wants to wait for 10 or more seconds for search results to appear). So to reliably detect all images used by Realblog all pages would need to be retrieved from the database, passed to `evaluate_scripting()`, and then somehow the images in use need would be detected (what it not trivial per se nowadays, since not only `<img>` elements are relevant any longer). Not even sure if the regex in the filebrowser is up to date:

https://github.com/cmsimple-xh/cmsimple-xh/blob/f3bb96dacf73370adb7dcc10fc503fc646235d12/plugins/filebrowser/classes/Controller.php#L194-L199

Perhaps we should just document the issue (and also the related search issue).

**Comment by olape-git** [comment]
> Not even sure if the regex in the filebrowser is up to date:

We have made an effort. But have we considered all the possibilities?

Yes, it probably doesn't really make sense to use the API.
You could keep a kind of inventory directory for images and other files, listing which file is used by whom and where.
But that would also involve a considerable amount of work in the system and also in the plugins.


> Perhaps we should just document the issue (and also the related search issue).

Yes, that would certainly be good.
Perhaps it would also be a good idea to use a kind of recycle bin instead of the normal delete function.
This would make it possible to restore a file via the system.

**Comment by cmb69** [comment]
> Perhaps it would also be a good idea to use a kind of recycle bin instead of the normal delete function.
> This would make it possible to restore a file via the system.

My personal stance on `userfiles` is: those contents have been uploaded by the user (file browser, FTP, doesn't really matter how). So the user can keep everything on their local machine (or a cloud, an USB stick, doesn't really matter where), and restore from there (or even sync with a capable FTP client).

The only point I'm seeing in a recycle bin would be for quick fixes during a session: "let's delete this image – oh no, it is used somewhere – now I need to get it from my local machine (or some back-up, whatever)". Probably nice, but still some work to implement.

**Comment by olape-git** [comment]
> So the user can keep everything on their local machine (or a cloud, an USB stick, doesn't really matter where), and restore from there (or even sync with a capable FTP client).

Yes, of course, the idea is not intended as a backup.

> The only point I'm seeing in a recycle bin would be for quick fixes during a session: "let's delete this image – oh no, it is used somewhere – now I need to get it from my local machine (or some back-up, whatever)".

That was the idea behind it.
